### ISSUE TITLE
[inductor] Unrolled sum reduction produces wrong results for non-contiguous (transposed) inputs after `unsqueeze+expand`

### ISSUE THREAD

**Issue by XAheli** [pytorch/pytorch#189401]
### 🐛 Describe the bug

When `make_pointwise(ops.mul)` + `make_reduction("sum")` operates on non-contiguous (transposed) inputs after `unsqueeze` + broadcast `expand`, the generated Triton kernel uses **wrong stride multipliers** — it assumes contiguous strides for the logical shape instead of preserving the actual transposed strides.

The `ops.dot` + `reduction("dot")` path (native_matmul) is NOT affected because it bypasses the standard tiling heuristic and forces `{y: M, x: N, r0_: K}` tiling unconditionally.

### Root cause

Traced through 4 independent investigations to:

1. **Reduction unrolling** (`ir.py:1770-1787`): When `reduction_numel < unroll_reductions_threshold` (default 8) AND `reduction_type != "dot"`, the reduction is unrolled into a `Pointwise` node via `_unroll_reduction_fn`. The `"dot"` reduction is explicitly exempt.

2. **View.create path difference**: Without `clone`, the input is a `ReinterpretView` with `is_storage_and_layout=True`, so `View.create` uses the `_compute_stride` path. With `clone`, the input becomes a `Pointwise` with `is_storage_and_layout=False`, routing through the more robust `dynamic_reshape_indexer` (symbolic reindexing).

3. **Generated code evidence**: For `mm(64×3, (4×3).T)`, the buggy kernel loads `in_ptr1 + (4*k + x0)` (contiguous strides `(4,1)` for shape `(3,4)`) when the actual transposed strides are `(1,3)`. The correct load pattern is `in_ptr1 + (k + 3*x0)`.

Related to #186850 (closed as non-reproducible, different reproducer).

### Reproducer

```python
import torch

a = torch.randn(64, 3, device='cuda')
b = torch.randn(4, 3, device='cuda')

@torch.compile
def fn(x, y):
    return torch.mm(x, y.T)

ref = torch.mm(a, b.T)
out = fn(a, b)
print(f"max diff: {(ref - out).abs().max().item()}")
# Expected: ~1e-7 (float32 tolerance)
# Actual: ~8.0 (completely wrong)
```

**Note**: This only reproduces when mm is decomposed to `unsqueeze + mul + sum` at the lowering level (e.g., via the small-mm pointwise optimization in PR #189149). Standard cuBLAS mm is not affected.

To reproduce without PR #189149, insert this in `tuned_mm` after `mm_args`:
```python
mat1 = lowerings[aten.unsqueeze](mat1, -1)
mat2 = lowerings[aten.unsqueeze](mat2, 0)
args, _ = transform_args(args=[mat1, mat2], kwargs={}, broadcast=True,
                          type_promotion_kind=None, convert_input_to_bool=False)
return make_reduction("sum")(make_pointwise(ops.mul)(*args), axis=1)
```

### Workaround

`clone` before `unsqueeze` forces the input through the `dynamic_reshape_indexer` path:
```python
mat1 = L.clone(mat1)
mat2 = L.clone(mat2)
```
The clone has zero runtime cost because the scheduler fuses the clone's Pointwise node into the downstream mul kernel.

### Versions

PyTorch 2.14.0a0+git30e161d (viable/strict), CUDA 12.8, H200

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @jataylo @jansel @eellison @yanboliang @ngimel
